syntax = "proto3";

package wallet;

service WalletService {
  rpc MakeWallet(MakeWalletRequest) returns(MakeWalletResponse);
}

message MakeWalletRequest {
  enum EthNetwork {
    MAINNET = 0;
    TESTNET_MEDALLA = 1;
  }

  string password = 1;
  uint32 numKeysToGenerate = 2;
  EthNetwork network = 3;
  uint64 amount = 4;
}

message MakeWalletResponse {
  message Credential {
    string deposit_data = 1;
    string signing_keystore = 2;
  }

  string mnemonic = 2;
  repeated Credential credentials = 3;
}

